python/ray/serve/utils.py

```diff
@@ -313,6 +314,53 @@ def parse_import_path(import_path: str):
     return ".".join(nodes[:-1]), nodes[-1]
 
 
+def merge_runtime_envs(parent_env: Dict, child_env: Dict) -> Dict:
+    """Creates a runtime_env dict by merging a parent and child environment.
+
+    This method is not destructive. It leaves the parent and child envs
+    the same.
+
+    The merge is a shallow update where the child environment inherits the
+    parent environment's settings. If the child environment specifies any
+    env settings, those settings take precdence over the parent.
+        - Note: env_vars are a special case. The child's env_vars are combined
```

Comment:
can you explain why btw?

Reply:
Sure, the special casing comes from the current [`runtime_env` inheritance rules](https://docs.ray.io/en/releases-1.12.1/ray-core/handling-dependencies.html#inheritance). From that documentation:

> The `runtime_env["env_vars"]` field will be merged with the `runtime_env["env_vars"]` field of the parent. This allows for environment variables set in the parent’s runtime environment to be automatically propagated to the child, even if new environment variables are set in the child’s runtime environment.